### PR TITLE
Add persistence

### PR DESCRIPTION
**Pull request by tekknolagi**
Can write the entire DB to git for later!

### CODE REVIEW

**Comment by ybur-yug** [comment]
Elegant and clean! Love it. This was right along the lines of what I was thinking of doing tomorrow morning. Mergin!

File: spec/gkv_spec.rb [hunk 1]
```diff
@@ -114,4 +114,20 @@ def clear_db
...
+  context "on save" do
+    it 'returns a hash' do
+      db.set('hello', 'world')
+      expect(db.save).to be_an_instance_of String
+    end
+
+    it 'can be loaded given a hash' do
+      db.set('hello', 'world')
+      hash = db.save
+      $ITEMS = {}
```
Comment (ybur-yug) [inline spec/gkv_spec.rb]: We could call the helper function defined at the top of spec here, but this is a very minor caveat. Still lgtm

**Comment by tekknolagi** [comment]
@ybur-yug yes I agree with you completely. Will refactor.

**Comment by tekknolagi** [comment]
Oh shoot did you already merge?

**Comment by ybur-yug** [comment]
Yeah, its not a big deal though. We can squash it with another commit. Isn't complex enough it affects the history's readability

**Pull request merged**